osmomath/exp2_test.go

```diff
@@ -0,0 +1,296 @@
+			// https://www.wolframalpha.com/input?i=2%5E0.3334567+37+digits
+			expectedResult: osmomath.MustNewDecFromStr("1.260028791934303989065848870753742298"),
+
+			errTolerance: osmoutils.ErrTolerance{
+				AdditiveTolerance:       sdk.MustNewDecFromStr("0.00000000000000007"),
+				MultiplicativeTolerance: minDecTolerance.Mul(sdk.NewDec(10)),
+				RoundingDir:             osmomath.RoundDown,
+			},
+		},
+		"exp2(0.84864288)": {
+			exponent: osmomath.MustNewDecFromStr("0.84864288"),
+			// https://www.wolframalpha.com/input?i=2%5E0.84864288+37+digits
+			expectedResult: osmomath.MustNewDecFromStr("1.800806138872630518880998772777747572"),
+
+			errTolerance: osmoutils.ErrTolerance{
+				AdditiveTolerance:       sdk.MustNewDecFromStr("0.00000000000000002"),
+				MultiplicativeTolerance: minDecTolerance.Mul(sdk.NewDec(10)),
+				RoundingDir:             osmomath.RoundUnconstrained, // TODO: confirm if this is acceptable.
+			},
+		},
+		"exp2(0.999999999999999999999999999999999956)": {
+			exponent: osmomath.MustNewDecFromStr("0.999999999999999999999999999999999956"),
+			// https://www.wolframalpha.com/input?i=2%5E0.999999999999999999999999999999999956+37+digits
+			expectedResult: osmomath.MustNewDecFromStr("1.999999999999999999999999999999999939"),
+
+			errTolerance: osmoutils.ErrTolerance{
+				AdditiveTolerance:       minDecTolerance,
+				MultiplicativeTolerance: minDecTolerance,
+				RoundingDir:             osmomath.RoundDown,
+			},
+		},
+		// out of bounds.
+		"exponent < 0 - panic": {
+			exponent:    osmomath.ZeroDec().Sub(smallValue),
+			expectPanic: true,
+		},
+		"exponent > 1 - panic": {
+			exponent:    osmomath.OneDec().Add(smallValue),
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			osmoassert.ConditionalPanic(t, tc.expectPanic, func() {
+				// System under test.
+				result := osmomath.Exp2ChebyshevRationalApprox(tc.exponent)
+
+				// Reuse the same test cases for exp2 that is a wrapper around Exp2ChebyshevRationalApprox.
+				// This is done to reduce boilerplate from duplicating test cases.
+				resultExp2 := osmomath.Exp2(tc.exponent)
+				require.Equal(t, result, resultExp2)
+
+				require.Equal(t, 0, tc.errTolerance.CompareBigDec(tc.expectedResult, result))
+			})
+		})
+	}
+}
+
+func TestExp2(t *testing.T) {
+	tests := map[string]struct {
+		exponent       osmomath.BigDec
+		expectedResult osmomath.BigDec
+		errTolerance   osmoutils.ErrTolerance
+		expectPanic    bool
+	}{
+		"exp2(28.5)": {
+			exponent: osmomath.MustNewDecFromStr("28.5"),
+			// https://www.wolframalpha.com/input?i=2%5E%2828.5%29+45+digits
+			expectedResult: osmomath.MustNewDecFromStr("379625062.497006211556423566253288543343173698"),
+
+			errTolerance: osmoutils.ErrTolerance{
+				AdditiveTolerance:       minDecTolerance,
+				MultiplicativeTolerance: minDecTolerance,
+				RoundingDir:             osmomath.RoundUnconstrained, // TODO: confirm if this is acceptable.
```

Comment:
My answer all for all these "are they acceptable" is Yes!

Just need comments on the internal exp and overall exponent that

"The answer is correct up to a factor of 10^-22. Meaning that `actual = k * returned` for `k in [1 - 10^(-22), 1+ 10^22]`

Reply:
We can say 10^(-18) in godoc since thats what we test for, just add a note to ourself that we actually expect 10^22.

Reply:
godoc added